File: tks/RestAdapters/src/main/kotlin/adapters/LibraryItemResourceAdapter.java
package adapters;

import dto.LibraryItemDto;
import exceptions.ResourceNotFoundException;
import exceptions.TypeValidationFailedException;
import exceptions.UnknownResourceException;
import mappers.LibraryItemMapper;
import model.values.AccessionNumber;
import ports.primary.IResourceService;

import javax.inject.Inject;

public class LibraryItemResourceAdapter {

    @Inject
    private IResourceService resourceService;
    @Inject
    private LibraryItemMapper mapper;

    public void add(LibraryItemDto dto) throws UnknownResourceException {
        var resource = mapper.toDomainObject(dto);
        resourceService.create(resource);
    }

    public LibraryItemDto query(String id) throws TypeValidationFailedException, ResourceNotFoundException {
        var accessionNumber = new AccessionNumber(id);
        var resource = resourceService.getDetails(accessionNumber);
        return mapper.toDto(resource);
    }
}

File: tks/Microservices/rent-service/src/main/java/it/p/lodz/pl/service/rents/adapters/LibraryItemResourceAdapter.kt
package it.p.lodz.pl.service.rents.adapters

import domain.exceptions.ResourceNotFoundException
import domain.exceptions.TypeValidationFailedException
import domain.exceptions.UnknownResourceException
import domain.model.values.AccessionNumber
import it.p.lodz.pl.service.rents.dto.LibraryItemDto
import it.p.lodz.pl.service.rents.mappers.LibraryItemMapper
import ports.primary.combined.IResourceService
import javax.inject.Inject

class LibraryItemResourceAdapter @Inject constructor(
    private val resourceService: IResourceService,
    private val mapper: LibraryItemMapper
) {

    @Throws(UnknownResourceException::class)
    fun add(dto: LibraryItemDto?): String {
        val resource = mapper.toDomainObject(dto!!)
        resourceService.create(resource)
        return resource.accessionNumber!!.value
    }

    @Throws(TypeValidationFailedException::class, ResourceNotFoundException::class)
    fun query(id: String?): LibraryItemDto {
        val accessionNumber = AccessionNumber(id!!)
        val resource = resourceService.getDetails(accessionNumber)
        return mapper.toDto(resource!!)
    }
}